snappy/click.go

```diff
@@ -214,6 +220,47 @@ ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.Target}} "$@"
 	return templateOut.String(), nil
 }
 
+func generateSnapDesktopFile(binary Binary, pkgPath string, m *packageYaml) (string, error) {
+	desktopTemplate := `[Desktop Entry]
+Encoding=UTF-8
+Type=Application
+Name={{.Name}}
+Comment={{.Comment}}
+Icon={{.Icon}}
+Exec={{.Exec}}
```

Comment:
This doesn't really work as the exec line in the YAML is designed for command line programs, not GUI ones. For instance where to put files or URLs in the exec line is needed.